src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -443,4 +441,132 @@ export class CourseExerciseDetailsComponent extends AbstractScienceComponent imp
     changeExampleSolution() {
         this.exampleSolutionCollapsed = !this.exampleSolutionCollapsed;
     }
+
+    // INSTRUCTOR ACTIONS
+
+    createInstructorActions() {
+        if (this.exercise?.isAtLeastTutor) {
+            this.instructorActionItems = this.createTutorActions();
+        }
+        if (this.exercise?.isAtLeastEditor) {
+            this.instructorActionItems.push(...this.createEditorActions());
+        }
+        if (this.exercise?.isAtLeastInstructor && this.QUIZ_ENDED_STATUS.includes(this.quizExerciseStatus)) {
+            this.instructorActionItems.push(this.getReEvaluateItem());
+        }
+    }
+
+    createTutorActions(): InstructorActionItem[] {
+        const tutorActionItems = [...this.getDefaultItems()];
+        if (this.exercise?.type === ExerciseType.QUIZ) {
+            tutorActionItems.push(...this.getQuizItems());
+        } else {
+            tutorActionItems.push(this.getParticipationItem());
+        }
+        return tutorActionItems;
+    }
+
+    getDefaultItems(): InstructorActionItem[] {
+        return [
+            {
+                routerLink: `${this.baseResource}`,
+                icon: faEye,
+                translation: 'entity.action.view',
+            },
+            {
+                routerLink: `${this.baseResource}scores`,
+                icon: faTable,
+                translation: 'entity.action.scores',
+            },
+        ];
+    }
+
+    getQuizItems(): InstructorActionItem[] {
+        return [
+            {
+                routerLink: `${this.baseResource}preview`,
+                icon: faEye,
+                translation: 'artemisApp.quizExercise.preview',
+            },
+            {
+                routerLink: `${this.baseResource}solution`,
+                icon: faEye,
+                translation: 'artemisApp.quizExercise.solution',
+            },
+        ];
+    }
+
+    getParticipationItem(): InstructorActionItem {
+        return {
+            routerLink: `${this.baseResource}participations`,
+            icon: faListAlt,
+            translation: 'artemisApp.exercise.participations',
+        };
+    }
+
+    createEditorActions(): InstructorActionItem[] {
+        const editorItems: InstructorActionItem[] = [];
+        if (this.exercise?.type === ExerciseType.QUIZ) {
+            editorItems.push(this.getStatisticItem('quiz-point-statistic'));
+        }
+        if (this.exercise?.type === ExerciseType.MODELING) {
+            editorItems.push(this.getStatisticItem('exercise-statistics'));
+        }
+        if (this.exercise?.type === ExerciseType.PROGRAMMING) {
+            editorItems.push(this.getGradingItem());
+        }
+        if (this.QUIZ_EDITABLE_STATUS.includes(this.quizExerciseStatus)) {
+            editorItems.push(this.getQuizEditItem());
+        }
```

Comment:
```suggestion
        if (this.exercise?.type === ExerciseType.QUIZ) {
            editorItems.push(this.getStatisticItem('quiz-point-statistic'));
            if (this.QUIZ_EDITABLE_STATUS.includes(this.quizExerciseStatus)) {
                editorItems.push(this.getQuizEditItem());
            }
        }
        else if (this.exercise?.type === ExerciseType.MODELING) {
            editorItems.push(this.getStatisticItem('exercise-statistics'));
        }
        else if (this.exercise?.type === ExerciseType.PROGRAMMING) {
            editorItems.push(this.getGradingItem());
        }
```
Maybe personal preference but it is more intuitive for me what is checked when and what cases can happen at the same time